hs-bindgen/src/HsBindgen/Hs/AST.hs

```diff
@@ -98,6 +100,15 @@ data Struct (n :: Nat) = Struct {
     }
   deriving stock (Eq, Generic, Show)
 
+instance GEq Struct where
+  geq :: forall n m. Struct n -> Struct m -> Maybe (n :~: m)
+  geq sL sR = Vec.withDict (structFields sL) $ Vec.withDict (structFields sR) $
+    case Nat.eqNat @n @m of
+      Just Refl
+        | sL == sR  -> Just Refl
+        | otherwise -> Nothing
+      Nothing       -> Nothing
+
```

Comment:
With the `ApEq` newtype I suggested above, this could simply be

```hs
data Struct (n :: Nat) = ...
  deriving stock (Eq, Generic, Show)
  deriving GEq via ApEq Struct
```